issues: add PostRequest

tbg opened the pull request:
The list of parameters required to post an issue was getting out of
hand. I'm going to have to touch this code soon, so might as well
clean it up.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/41807)
<!-- Reviewable:end -->

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/41807#-:-LrnyHIE4N2OHagxw-x0:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

___
*[pkg/cmd/internal/issues/issues.go, line 420 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/41807#-Lrny2ou8eJtITDmRh9D:-Lrny2ou8eJtITDmRh9E:bx6vdqf) ([raw file](https://github.com/cockroachdb/cockroach/blob/86406b0628e230160c6e537a698d6bb000c206cc/pkg/cmd/internal/issues/issues.go#L420)):*
> ```Go
> // test failure.
> type PostRequest struct {
> 	Title, PackageName, TestName, Message, Artifacts, AuthorEmail string
> ```

pls comment how Artifacts is used. And what fields are optional, if you can.


<!-- Sent from Reviewable.io -->

tbg committed:
issues: add PostRequest

The list of parameters required to post an issue was getting out of
hand. I'm going to have to touch this code soon, so might as well
clean it up.

Release note: None

tbg commented:
Done. TFTR!
